src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/repository/RepositoryService.kt

```diff
@@ -33,6 +44,80 @@ class RepositoryService(@Value("\${mordor.root_path}") private val rootPathStr:
         return fullPath
     }
 
+    fun getEntity(entityPath: String): RepositoryEntity? {
+        val path = getAbsolutePath(entityPath)
+        return getRepositoryEntity(path)
+    }
+
+    fun fileExists(path: String): Boolean {
+        return getAbsolutePath(path).toFile().exists()
+    }
+
+    fun saveFile(path: String, inputStreamSource: InputStreamSource): RepositoryEntity? {
+        val inputStream = inputStreamSource.inputStream
+        val outputPath = getAbsolutePath(path)
+
+        outputPath.toFile().mkdirs()
+        val outputStream = FileOutputStream(outputPath.toString())
+        outputStream.write(inputStream.readBytes())
+        entryCreator.create(outputPath.toFile())
+
+        return getEntity(outputPath.toString())
+    }
+
+    @Transactional
+    fun move(from: String, to: String) {
+        val absoluteToPath = getAbsolutePath(to)
+        val entity = getEntity(from)
+        val absoluteFromPath = getAbsolutePath(from)
+
+        if (entity is RepositoryFile) {
+            Files.move(absoluteFromPath, absoluteToPath)
+            moveEntry(absoluteToPath, absoluteFromPath)
+        } else {
```

Comment:
if else entity is RepositoryDirectory

(we might introduce a new entity one day)